File: packages/java/anip-service/src/main/java/dev/anip/service/StreamEvent.java
package dev.anip.service;

import java.util.Map;

/**
 * A single event in a streaming invocation.
 */
public class StreamEvent {

    private final String type;    // "progress", "completed", "failed"
    private final Map<String, Object> payload;

    public StreamEvent(String type, Map<String, Object> payload) {
        this.type = type;
        this.payload = payload;
    }

    public String getType() {
        return type;
    }

    public Map<String, Object> getPayload() {
        return payload;
    }
}

File: packages/csharp/src/Anip.Service/StreamEvent.cs
namespace Anip.Service;

/// <summary>
/// Represents a single SSE event in a streaming invocation.
/// </summary>
public class StreamEvent
{
    /// <summary>Event type: "progress", "completed", or "failed".</summary>
    public string Type { get; set; } = "";

    /// <summary>Full SSE data payload.</summary>
    public Dictionary<string, object?> Payload { get; set; } = new();
}
